test/util/util.go

```diff
@@ -1250,6 +1251,56 @@ func MustCreate(ctx context.Context, c client.Client, obj client.Object) {
 	gomega.Expect(c.Create(ctx, obj)).Should(gomega.Succeed())
 }
 
+func IsPodReady(pod *corev1.Pod) bool {
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == corev1.PodReady {
+			return cond.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+// WaitForWebhookEndpointsReady waits for the webhook service EndpointSlice
+// to contain the current controller pod IPs before making webhook requests.
+func WaitForWebhookEndpointsReady(ctx context.Context, k8sClient client.Client) {
```

Comment:
nit, move this function back to `test/util/e2e.go`. I don't see us using it outside of e2e tests. And I would suggest to unexport it.